app/src/org/commcare/utils/FileUtil.java

```diff
@@ -449,6 +449,13 @@ public static String getExtension(String filePath) {
         return "";
     }
 
+    public static String getFileName(String filePath) {
+        if (filePath.contains("/")) {
+            return last(filePath.split("/"));
+        }
+        return "";

```

Comment:
This doesn't seem right... wouldn't this result in the file's path being the file path of the parent directory (and hence overwriting that directory)?